packages/bff/src/services/purposeService.ts

```diff
@@ -1,19 +1,260 @@
-import { WithLogger } from "pagopa-interop-commons";
+import { WithLogger, toSetToArray } from "pagopa-interop-commons";
 import {
   PurposeId,
   PurposeVersionId,
   unsafeBrandId,
 } from "pagopa-interop-models";
-import { bffApi } from "pagopa-interop-api-clients";
-import { PagoPAInteropBeClients } from "../providers/clientProvider.js";
-import { purposeNotFound } from "../model/domain/errors.js";
-import { BffAppContext } from "../utilities/context.js";
+import { bffApi, purposeApi, tenantApi } from "pagopa-interop-api-clients";
+import {
+  AgreementProcessClient,
+  AuthorizationProcessClient,
+  CatalogProcessClient,
+  PurposeProcessClient,
+  TenantProcessClient,
+} from "../providers/clientProvider.js";
+import {
+  agreementNotFound,
+  eServiceDescriptorNotFound,
+  eServiceNotFound,
+  purposeDraftVersionNotFound,
+  purposeNotFound,
+  tenantNotFound,
+} from "../model/domain/errors.js";
+import { BffAppContext, Headers } from "../utilities/context.js";
+import { toBffApiCompactClient } from "../model/domain/apiConverter.js";
+import { isUpgradable } from "../model/modelMappingUtils.js";
+import { getLatestAgreement } from "./agreementService.js";
+import { getAllClients } from "./clientService.js";
+
+export const getCurrentVersion = (
+  purposeVersions: purposeApi.PurposeVersion[]
+): purposeApi.PurposeVersion | undefined => {
+  const statesToExclude: purposeApi.PurposeVersionState[] = [
+    purposeApi.PurposeVersionState.Values.WAITING_FOR_APPROVAL,
+    purposeApi.PurposeVersionState.Values.REJECTED,
+  ];
+  return purposeVersions
+    .filter((v) => !statesToExclude.includes(v.state))
+    .sort(
+      (a, b) =>
+        new Date(a.createdAt).getTime() - new Date(b.createdAt).getTime()
+    )
+    .at(-1);
+};
+
+// eslint-disable-next-line max-params
+async function getPurposes(
+  {
+    purposeClient,
+    eserviceClient,
+    tenantClient,
+    agreementClient,
+    authorizationClient,
+  }: {
+    purposeClient: PurposeProcessClient;
+    eserviceClient: CatalogProcessClient;
+    tenantClient: TenantProcessClient;
+    agreementClient: AgreementProcessClient;
+    authorizationClient: AuthorizationProcessClient;
+  },
```

Comment:
we should have a [specific type](https://github.com/pagopa/interop-be-monorepo/blob/f82607bcb8ced2768014e25a7a746a63c590c110/packages/bff/src/providers/clientProvider.ts#L37) for this objects, in this case there are a lots clients.

```suggestion
 clients : PagoPAInteropBeClients,
```